proposals/action-on-user-settings-changed-api.md

```diff
@@ -0,0 +1,68 @@
+# Extension API Proposal: action.onUserSettingsChanged
+
+## Background
+
+Proposal based on [issue #346](https://github.com/w3c/webextensions/issues/346): New API: UserSettings.isOnToolbar change event
+
+## API Overview
+
+### Objective
+
+Fire an event when user-specified settings relating to an extension's action changed.
+
+### Use Cases
+
+#### Pinning / unpinning extension
+
+Extensions can know if its action is pinned by calling `chrome.action.getUserSettings.` If the extension wants to know when this value changes, it has to poll the API method. Instead, it could register for an event.
+
+## Implementation
+
+### API Schema
+
+#### Types
+
+```javascript
+dictionary UserSettings {
+  isOnToolbar: boolean
+}
+```
+
+#### Events
+
+```javascript
+// Fires when user-specified settings relating to an extension's action changed.
+<browser>.action.onUserSettingsChanged.addListener(
+  callback: function, // where callback looks like: (userSettings: UserSettings) => void
+);
+
+// Deregisters the event listener.
+<browser>.action.onUserSettingsChanged.removeListener(
+  callback: function, // where callback looks like: (userSettings: UserSettings) => void
+);
+```
+
+### New Permissions
+
+| Permission Added | Proposed Warning |
+| ---------------- | ------------------------------------------------------- |
+| N/A              | This is an addition to the action API, which is exposed if the extension includes the "action" key in the manifest. |
+
+### Design and Implementation
+
+Currently, `UserSettings` only has one property. This makes things easy, as we just need to trigger the event when such property changes. If new properties are added, we will trigger the event every time any of the properties change.
```

Comment:
Could you explicitly call out whether the event will receive an object with the modified values, or an object with the same value as `getUserSettings`? See my other comment at the API signature for more context.

Reply:
Done.